encode_test.go

```diff
@@ -509,46 +508,68 @@ var marshalTests = []struct {
 	},
 }
 
-func (s *S) TestMarshal(c *C) {
+func TestMarshal(t *testing.T) {
 	defer os.Setenv("TZ", os.Getenv("TZ"))
 	os.Setenv("TZ", "UTC")
 	for i, item := range marshalTests {
-		c.Logf("test %d: %q", i, item.data)
-		data, err := yaml.Marshal(item.value)
-		c.Assert(err, IsNil)
-		c.Assert(string(data), Equals, item.data)
+		t.Run(fmt.Sprintf("test %d: %q", i, item.data), func(t *testing.T) {
+			data, err := yaml.Marshal(item.value)
+			if err != nil {
+				t.Fatalf("Marshal() returned error: %v", err)
+			}
+			if string(data) != item.data {
+				t.Fatalf("Marshal() returned\n%q\nbut expected\n%q", string(data), item.data)
+			}
+		})
 	}
 }
 
-func (s *S) TestEncoderSingleDocument(c *C) {
+func TestEncoderSingleDocument(t *testing.T) {
 	for i, item := range marshalTests {
-		c.Logf("test %d. %q", i, item.data)
-		var buf bytes.Buffer
-		enc := yaml.NewEncoder(&buf)
-		err := enc.Encode(item.value)
-		c.Assert(err, Equals, nil)
-		err = enc.Close()
-		c.Assert(err, Equals, nil)
-		c.Assert(buf.String(), Equals, item.data)
+		t.Run(fmt.Sprintf("test %d. %q", i, item.data), func(t *testing.T) {
+			var buf bytes.Buffer
+			enc := yaml.NewEncoder(&buf)
+			err := enc.Encode(item.value)
+			if err != nil {
+				t.Fatalf("Encode() returned error: %v", err)
+			}
+			err = enc.Close()
+			if err != nil {
+				t.Fatalf("Close() returned error: %v", err)
+			}
+			if buf.String() != item.data {
+				t.Fatalf("Encode() returned\n%q\nbut expected\n%q", buf.String(), item.data)
+			}
+		})
 	}
 }
 
-func (s *S) TestEncoderMultipleDocuments(c *C) {
+func TestEncoderMultipleDocuments(t *testing.T) {
 	var buf bytes.Buffer
 	enc := yaml.NewEncoder(&buf)
 	err := enc.Encode(map[string]string{"a": "b"})
-	c.Assert(err, Equals, nil)
+	if err != nil {
+		t.Fatalf("Encode() returned error: %v", err)
+	}
 	err = enc.Encode(map[string]string{"c": "d"})
-	c.Assert(err, Equals, nil)
+	if err != nil {
+		t.Fatalf("Encode() returned error: %v", err)
+	}
 	err = enc.Close()
-	c.Assert(err, Equals, nil)
-	c.Assert(buf.String(), Equals, "a: b\n---\nc: d\n")
+	if err != nil {
+		t.Fatalf("Close() returned error: %v", err)
+	}
+	if buf.String() != "a: b\n---\nc: d\n" {
+		t.Fatalf("Encode() returned\n%q\nbut expected\n%q", buf.String(), "a: b\n---\nc: d\n")
+	}
 }
 
-func (s *S) TestEncoderWriteError(c *C) {
+func TestEncoderWriteError(t *testing.T) {
 	enc := yaml.NewEncoder(errorWriter{})
 	err := enc.Encode(map[string]string{"a": "b"})
-	c.Assert(err, ErrorMatches, `yaml: write error: some write error`) // Data not flushed yet
+	if err == nil || !strings.Contains(err.Error(), `yaml: write error: some write error`) {
+		t.Fatalf("Encode() returned %v, want error containing %q", err, `yaml: write error: some write error`)
+	}
```

Comment:
I would introduce a variable here


```suggestion
	expected := `yaml: write error: some write error`
	if err == nil || !strings.Contains(err.Error(), expected) {
		t.Fatalf("Encode() returned %v, want error containing %q", err, expected)
	}
```

I'm a bit worried error messages detected and report may divert over the time with code changes

Reply:
Yeah, let me fix that. ⏳

Reply:
@ccoVeille would you mind taking a look at #51 ? :)

Reply:
I'll do in a separate MR.